portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+
+    try {
+      int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
+
+      jsonResponse.add("comments", commentsJson);
+      jsonResponse.addProperty("success", true);
+      } catch (NumberFormatException nfe) {
+        jsonResponse.addProperty("success", false);
+        String errorMessage = nfe.getMessage();
 
-    Gson gson = new Gson();
+        /* When NumberFormatException caught because
+        *  max-comments parameter is null */
+        if (errorMessage == "null") {
+          errorMessage = "Please enter a value for maxComments.";
+        }
+        else { /* When max-comments parameter is not a number */
+          errorMessage = "NumberFormatException: " + errorMessage +
+            ": Please enter a number for maxComments.";
+        }
+        jsonResponse.addProperty("errorMessage", errorMessage);
+      }
 
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(comments));
+    response.getWriter().println(jsonResponse.toString());
```

Comment:
The http response can be used to represent the error in a better manner. 

https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServletResponse.html#sendError(int,%20java.lang.String)

The client side code can do a better job using the code and treat the error condition in a different manner .

Reply:
I think this does not work for response.setContentType("application/json;"); So you do have to set the error message manually (see https://stackoverflow.com/q/20320074 for example). However you can still set the status code.